tests/pytests/integration/utils/test_smb.py

```diff
@@ -0,0 +1,378 @@
+"""
+Test utility methods that communicate with SMB shares.
+"""
+import contextlib
+import getpass
+import os
+import pathlib
+import shutil
+import signal
+import subprocess
+import tempfile
+import time
+
+import pytest
+
+import salt.utils.files
+import salt.utils.network
+import salt.utils.path
+import salt.utils.smb
+
+IPV6_ENABLED = bool(salt.utils.network.ip_addrs6(include_loopback=True))
+
+
+pytestmark = [
+    pytest.mark.skipif(
+        not salt.utils.smb.HAS_SMBPROTOCOL,
+        reason='"smbprotocol" needs to be installed.',
+    ),
+    pytest.mark.skip_if_binaries_missing("smbd", check_all=False),
+    pytest.mark.skip_unless_on_linux(reason="using Linux samba to test smb"),
+]
+
+
+def check_pid(pid):
+    try:
+        os.kill(pid, 0)
+    except OSError:
+        return False
+    else:
+        return True
+
+
+@pytest.fixture
+def smb_dict(tmp_path):
+    samba_dir = tmp_path / "samba"
+    samba_dir.mkdir(parents=True)
+    assert samba_dir.exists()
+    assert samba_dir.is_dir()
```

Comment:
```suggestion
```
Unnecessary, the code to create the directory was just ran.

Reply:
code changed